linalg/sparsemat.cpp

```diff
@@ -702,7 +707,15 @@ void SparseMatrix::AddMult(const Vector &x, Vector &y, const double a) const
          CuMemAlloc(&dBuffer, bufferSize);
       }
 
-#if CUDA_VERSION > 10010 || CUDA_VERSION == 10010
+#if CUDA_VERSION > 10020 || CUDA_VERSION == 10020
```

Comment:
```suggestion
#if CUDA_VERSION > 11020 || CUDA_VERSION == 11020
```

Reply:
A laborious search in the cuda documentation _almost_ proves @YohannDudouit right.
Per documentation, `CUSPARSE_CSRMV_ALG1` was deprecated in `11.2.1`, when `CUSPARSE_SPMV_CSR_ALG1` was introduced. **It seems this should not work with 11.2.0, but only with 11.2.1 and higher**.
See:
https://docs.nvidia.com/cuda/archive/11.2.0/cusparse/index.html#cusparse-generic-api-reference
https://docs.nvidia.com/cuda/archive/11.2.1/cusparse/index.html#cusparse-generic-api-reference